provider/maas/devices.go

```diff
@@ -343,6 +343,11 @@ func (env *maasEnviron) createAndPopulateDevice(params deviceCreatorParams) (gom
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	defer func() {
+		if err != nil {
```

Comment:
The err here needs to refer to the error return value rather than just the err local variable. Otherwise the delete won't be done for the error return on line 419, because the function-local-scope err will be nil, even though the if-scope err is non-nil.

Reply:
this seems to have landed while you were making this comment. should we wait to mark the bug fixed until this is addressed?